### PR TITLE
[FEATURE] New `area_opacity` option in time series panel

### PR DESCRIPTION
**Pull request by sjcobb**
## Overview

Add optional `area_opacity` option to TimeSeriesChart panel. This includes a MUI Slider to customize the ECharts property [series-line.areaStyle.opacity](https://echarts.apache.org/en/option.html#series-line.areaStyle.opacity)

## TODOs For Follow Up PRs

- I started to add a visual regression test, but I'm planning to come back to it since I have some questions about mocking the range query response, see [branch here](https://github.com/perses/perses/compare/sjcobb/time-series-area-opacity-option...sjcobb/time-series-area-opacity-option-test) and [screenshot](https://github.com/perses/perses/pull/962#discussion_r1109928450)
- [Stacked area](https://echarts.apache.org/examples/en/editor.html?c=area-stack) support by adding new `stacking` visual option
  - schema will most likely be: 
```cue
#visual: {
	line_width?:   number & >=0.25 & <=3
	point_radius?: number & >=0 & <=6
	area_opacity?: number & >=0 & <=1
	stacking?: "Normal" | "Percent"
        // TBD: another option will be needed to control ordering
}
```

## Screenshots

panel_edit_area_opacity_slider

https://user-images.githubusercontent.com/9369625/219121680-07c0c621-4838-4d47-aac6-053eba292070.mp4

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).
- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.

### CODE REVIEW

**Comment by sjcobb** [comment]
Closing in favor of `area_shading: boolean` approach in #964 which uses a MUI Switch so that we have more control over display.

File: ui/panels-plugin/src/plugins/time-series-chart/time-series-chart-model.ts [hunk 1]
```diff
@@ -54,9 +55,12 @@ export const DEFAULT_LINE_WIDTH = 1.5;
 
 export const DEFAULT_POINT_RADIUS = 4;
 
+export const DEFAULT_AREA_OPACITY = 0;
+
 export const DEFAULT_VISUAL: VisualOptions = {
   line_width: DEFAULT_LINE_WIDTH,
   point_radius: DEFAULT_POINT_RADIUS,
+  area_opacity: DEFAULT_AREA_OPACITY,
```
Comment (sjcobb) [inline ui/panels-plugin/src/plugins/time-series-chart/time-series-chart-model.ts:63]: I started adding a new visual regression test for this new option and other visual / y_axis options, but I need to follow up with @juliepagano about a question with mocking the range query response, we may need to adjust some the types:

image

**Review by eunicorn — APPROVED** [review]

**Comment by juliepagano** [comment]
I noticed some weirdness where sometimes the opacity flickers when you have two lines and move your mouse around nearish the bottom line. Not sure if this a bug, intended behavior, or some third thing. Probably something that could be addressed in a follow-up either way.

flicker-opacity

File: ui/panels-plugin/src/plugins/time-series-chart/time-series-chart-model.ts (hunk 1 above)
Comment (juliepagano) [reply ui/panels-plugin/src/plugins/time-series-chart/time-series-chart-model.ts:63]: FYI for others that we chatted about this offline. Lemme know if you need any extra help on this when you do it in a follow-up.

File: ui/panels-plugin/src/plugins/time-series-chart/VisualOptionsEditor.tsx [hunk 2]
```diff
@@ -69,6 +83,21 @@ export function VisualOptionsEditor({ value, onChange }: VisualOptionsEditorProp
           />
         }
       />
+      <OptionsEditorControl
+        label={VISUAL_CONFIG.area_opacity.label}
+        control={
+          <Slider
+            data-testid={VISUAL_CONFIG.area_opacity.testId}
```
Comment (juliepagano) [inline ui/panels-plugin/src/plugins/time-series-chart/VisualOptionsEditor.tsx:90]: I wonder if you need the `data-testid`. When things have pretty clear labels and other a11y bits, you often don't need it.

**Review by juliepagano — APPROVED** [review]
One small question and a UX bit I dropped as a top-level comment. Looks good! 🚀

**Comment by sjcobb** [comment]
> I noticed some weirdness where sometimes the opacity flickers when you have two lines and move your mouse around nearish the bottom line.

Yeah this is definitely odd and not intended, I adjusted an [emphasis](https://echarts.apache.org/en/option.html#series-line.emphasis.disabled) property to account for it

**Pull request merged**